cupy/manipulation/add_remove.py

```diff
@@ -12,8 +13,62 @@
 
 # TODO(okuta): Implement resize
 
+_first_nonzero_krnl = core.ReductionKernel(
+    'raw S data, T indices, T len',
+    'T y',
+    'data[_j] ? _j : len',
+    'min(a, b)',
+    'y = a',
+    'len',
+    'first_nonzero'
+)
+_first_nonzero_cplx_krnl = core.ReductionKernel(
+    'raw S real, raw S imag, T indices, T len',
+    'T y',
+    'real[_j] || imag[_j] ? _j : len',
+    'min(a,b)',
+    'y = a',
+    'len',
+    'first_nonzero_cplx'
+)
+
+
+def trim_zeros(filt, trim='fb'):
+    """Trim the leading and/or trailing zeros from a 1-D array or sequence.
+
+    Returns the trimmed array
 
-# TODO(okuta): Implement trim_zeros
+    Args:
+        filt(1D array or sequence): Input array
+        trim(str, optional):
+            'fb' default option trims the array from both sides.
+            'f' option trim zeros from front.
+            'b' option trim zeros from front
+
+    Returns:
+        cupy.ndarray: trimmed input
+
+    .. seealso:: :func:`numpy.trim_zeros`
+    """
+    if filt.ndim > 1:
+        raise ValueError("Multi-dimensional trim is not supported")
+    start, end = cupy.asarray([0, filt.size - 1], dtype=int)
+    indices = cupy.arange(filt.size)
+    trim = trim.upper()
+    if 'F' in trim:
+        if filt.dtype.kind == 'c':
+            start = _first_nonzero_cplx_krnl(
+                filt.real, filt.imag, indices, filt.size)
+        else:
+            start = _first_nonzero_krnl(filt, indices, filt.size)
+    if 'B' in trim:
+        if filt.dtype.kind == 'c':
+            end = filt.size - _first_nonzero_cplx_krnl(
+                filt[::-1].real, filt[::-1].imag, indices, filt.size) - 1
+        else:
+            end = filt.size - _first_nonzero_krnl(
+                filt[::-1], indices, filt.size) - 1
+    return filt[start:end + 1]
```

Comment:
Please remove redundant `-1` and `+1`s.
```suggestion
    return filt[start:end]
```